src/core/data/ions/ions.h

```diff
@@ -244,52 +245,28 @@ namespace core
     template<typename Ions, typename GridLayout>
     void fixMomentGhosts(Ions& ions, GridLayout const& layout)
     {
-        if constexpr (Ions::dimension == 1)
+        using Mask = NdArrayMask;
+
+        for (auto& pop : ions)
         {
-            auto ix0 = layout.physicalStartIndex(QtyCentering::primal, Direction::X);
-            auto ix1 = layout.physicalEndIndex(QtyCentering::primal, Direction::X);
-            auto ix2 = layout.ghostEndIndex(QtyCentering::primal, Direction::X);
-
-            auto set = [](auto& pop, auto start, auto stop) {
-                for (auto i = start; i < stop; ++i)
-                {
-                    pop.density()(i) = NAN;
-                    for (auto& [id, type] : Components::componentMap)
-                        pop.flux().getComponent(type)(i) = NAN;
-                }
-            };
-
-
-            auto copy = [ix0, ix1](auto& pop) {
-                pop.density()(ix0 - 1) = pop.density()(ix0);
-                pop.density()(ix1 + 1) = pop.density()(ix1);
-
-                for (auto& [id, type] : Components::componentMap)
-                {
-                    pop.flux().getComponent(type)(ix0 - 1) = pop.flux().getComponent(type)(ix0);
-                    pop.flux().getComponent(type)(ix1 + 1) = pop.flux().getComponent(type)(ix1);
-                }
-            };
-
-            for (auto& pop : ions)
+            for (auto& [id, type] : Components::componentMap)
             {
-                set(pop, 0u, ix0 - 1); // leftGhostNodes
-                set(pop, ix1 + 1 + 1, ix2 + 1);
-                copy(pop);
-            }
-        }
-        else if constexpr (Ions::dimension == 2)
-        {
-            std::cout << "test2\n";
-        }
-        else if constexpr (Ions::dimension == 3)
-        {
-            std::cout << "test 3\n";
-        }
-    }
+                auto& flux_comp = pop.flux().getComponent(type);
+
+                auto nGhosts
+                    = layout.physicalStartIndex(flux_comp.physicalQuantity(), Direction::X);
```

Comment:
ok so the code says Direction::X but it is really valid for all directions since moments are primal primal primal and thus have the same number of ghosts in all directions.
But that is not obvious for a new reader or maybe us in some time.
I therefore would recommend adding a comment explaining that or changing the code so that Direction::X disappears

Reply:
comment https://github.com/PHAREHUB/PHARE/pull/325/files#diff-3c9be39f547148798f7a961790c9d328R246